widgetssdk/src/main/java/com/glia/widgets/core/visitor/GliaVisitorMediaRepository.java

```diff
@@ -1,21 +1,76 @@
 package com.glia.widgets.core.visitor;
 
+import com.glia.androidsdk.Engagement;
 import com.glia.androidsdk.comms.Media;
 import com.glia.androidsdk.comms.VisitorMediaState;
-import com.glia.widgets.di.Dependencies;
 
-public class GliaVisitorMediaRepository {
-    public interface VisitorMediaStateListener {
-        void onNewState(VisitorMediaState visitorMediaState);
+import java.util.ArrayList;
+import java.util.List;
+
+public class GliaVisitorMediaRepository implements VisitorMediaStateListener {
+    private final List<VisitorMediaStateListener> visitorMediaStateUpdates = new ArrayList<>();
+
+    private VisitorMediaState currentMediaState;
+    private boolean isOnHold;
+
+    public GliaVisitorMediaRepository() {
+        currentMediaState = null;
```

Comment:
they can be initialised together with the declarations, and get rid of constructor?